packages/lesswrong/lib/voting/reactions.tsx

```diff
@@ -0,0 +1,411 @@
+import keyBy from 'lodash/keyBy';
+
+export type NamesAttachedReactionType = {
+  /**
+   * Name of this reaction. Not shown directly, but typing its name into the
+   * reactions-search box will find it. Used as the key for identifying
+   * already-placed reactions.
+   */
+  name: string,
+  
+  /**
+   * Additional search terms that can be used to find a reaction. If the image
+   * is a mnemonic that doesn't match the name, should at least a name that
+   * describes the image.
+   */
+  searchTerms?: string[],
+  
+  /**
+   * Human-readable label string describing the reaction. Should start with a
+   * capital letter, and be short (ideally one word).
+   */
+  label: string,
+  
+  /**
+   * Image for this reaction. Should be an SVG, with the path relative to
+   * /public. See comment below for discussion of reaction image sources.
+   */
+  svg: string,
+  
+  /**
+   * A postprocessing filter to apply to the reaction's image. Can add padding
+   * (makes the image part smaller since the total size stays the same), and set
+   * opacity and saturation. Unset fields come from `defaultFilter`, below.
+   */
+  filter?: {
+    padding?: number,
+    opacity?: number,
+    saturate?: number,
+  },
+  
+  /**
+   * A description of this reaction, ideally about a sentence long. Optionally,
+   * expressed as a function that takes a content-type word, like "comment" or
+   * "post", so that you can make the description something like
+   * "This comment is <x>" and have it come out correct if it's a post/etc
+   * instead of a comment. (We don't yet support reactions on posts but we might
+   * in the future.
+   */
+  description: string|((contentType:string)=>string),
+}
+
+/**
+ * Reactions available.
+ *
+ * A reaction image should be an SVG image in /public/reactionImages. We don't
+ * get emojis directly from fonts because there's too much variability between
+ * platforms in how they look, but appropriately-licensed fonts like Noto Emoji
+ * and OpenMoji are good sources of images to use.
+ *
+ *     The Noun Project: https://thenounproject.com/
+ *     Noto Emoji: https://fonts.google.com/noto/specimen/Noto+Emoji
+ *     OpenMoji: https://openmoji.org/library/
+ *
+ * When you add a downloaded image, record where you got it in
+ * `public/reactionImages/sources.txt`. This is required to be able to meet the
+ * attribution requirement of many licenses. If you get an image from a source
+ * that hasn't been used before, you have to check its license and make sure
+ * it's suitable.
+ */
+
+// OpenClipart and OpenMoji
+/*export const namesAttachedReactions: NamesAttachedReactionType[] = [
```

Comment:
Does this need to be cleaned up?  There's a TODO at the bottom

Reply:
This is a previous palette in a different aesthetic, which became cruft when I went to Noun-Project-only. Removed.